internal/resources-validation.yml

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2019 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+jobs:
+# Spin up a Linux VM for validating code.
+- job: png_resources_validation
+  displayName: PNG Resources Validation
+  pool:
+    vmImage: 'ubuntu-latest'
+
+  steps:
+    - script: |
+        sudo apt update
+        sudo apt install -y pngcrush
+      displayName: 'Install Dependencies'
+    - script: |
+        # Print the full paths of PNG files
+        find . -type f -name "*.png" -print
+        # Run pngcrush command on PNG files and check for Profile issues
+        for file in $(find . -type f -name "*.png"); do
+          output=$(pngcrush -n -q "$file" 2>&1)
+          echo "$output"
+          if [[ "$output" == *"pngcrush: iCCP:"* ]]; then
+            echo "Warning processing file $file"
+            exit 1
```

Comment:
It would be helpful to list all warnings instead of returning when the first problem is found.

I don't know how to handle it here, maybe a flag to condition the `exit 1`? What do you think?

Reply:
I agree. You can have a counter starting at 0 and incrementing at each error.
Then you exit the counter: 
```python
sys.exit(min(counter,254))
```

Also, could you rename warning to error since it's a blocking step please?

Reply:
Sure,  you can see the new job logs from this commit 23371e0 build, while testing with an incorrect profile png.